File: project_windows/src/game/upgrade.c
/**
 * @file upgrade.c
 * @brief Weapon upgrade system implementation
 */

#include "upgrade.h"
#include "../core/inventory.h"
#include "../core/weapon.h"
#include "../data/registry.h"
#include "../utils/utils.h"
#include <stdio.h>
#include <string.h>

/* ============================================
 * INTERNAL HELPERS
 * ============================================ */

static const UpgradeRecipe *get_recipe_for_weapon(const Weapon *w) {
  if (!w)
    return NULL;
  return registry_get_upgrade(w->id, w->tier);
}

static int check_materials(const Inventory *inv, const UpgradeRecipe *recipe) {
  if (!inv || !recipe)
    return 0;

  for (int i = 0; i < recipe->material_count; i++) {
    int have = inventory_count_item(inv, recipe->materials[i].item_id);
    if (have < recipe->materials[i].quantity) {
      return 0;
    }
  }
  return 1;
}

static void consume_materials(Inventory *inv, const UpgradeRecipe *recipe) {
  if (!inv || !recipe)
    return;

  for (int i = 0; i < recipe->material_count; i++) {
    inventory_remove_item_by_id(inv, recipe->materials[i].item_id,
                                recipe->materials[i].quantity);
  }
}

/* ============================================
 * DISPLAY
 * ============================================ */

void upgrade_print_requirements(const Weapon *weapon, const Player *player) {
  if (!weapon)
    return;

  const UpgradeRecipe *recipe = get_recipe_for_weapon(weapon);

  if (!recipe) {
    if (weapon->tier >= weapon->max_tier) {
      printf("  This weapon is at maximum tier.\n");
    } else {
      printf("  No upgrade path available.\n");
    }
    return;
  }

  printf("  Upgrade: %s +%d -> +%d\n", weapon->name, weapon->tier - 1,
         recipe->to_tier - 1);
  printf("  Damage: %d -> %d (+%d)\n", weapon_get_damage(weapon),
         weapon_get_damage(weapon) + recipe->damage_bonus,
         recipe->damage_bonus);
  printf("\n  Requirements:\n");
  printf("    Gold: %d", recipe->gold_cost);
  if (player && player->gold >= recipe->gold_cost) {
    printf(" [OK]");
  } else {
    printf(" [NEED %d more]", recipe->gold_cost - (player ? player->gold : 0));
  }
  printf("\n");

  for (int i = 0; i < recipe->material_count; i++) {
    const ItemTemplate *item = registry_get_item(recipe->materials[i].item_id);
    const char *name = item ? item->name : recipe->materials[i].item_id;
    int have = player ? inventory_count_item(&player->inventory,
                                             recipe->materials[i].item_id)
                      : 0;
    int need = recipe->materials[i].quantity;

    printf("    %s: %d/%d", name, have, need);
    if (have >= need) {
      printf(" [OK]");
    } else {
      printf(" [NEED %d more]", need - have);
    }
    printf("\n");
  }
}

static void print_upgradeable_weapons(const Player *p) {
  printf("=== Upgradeable Weapons ===\n\n");

  int found = 0;

  /* Check equipped weapon */
  if (p->has_weapon && weapon_can_upgrade(&p->equipped_weapon)) {
    printf("  [E] ");
    weapon_print(&p->equipped_weapon);
    printf(" (EQUIPPED)\n");

    const UpgradeRecipe *recipe = get_recipe_for_weapon(&p->equipped_weapon);
    if (recipe) {
      int can_upgrade = p->gold >= recipe->gold_cost &&
                        check_materials(&p->inventory, recipe);
      printf("      -> Tier %d (%s)\n", recipe->to_tier,
             can_upgrade ? "READY" : "missing materials");
    }
    found = 1;
  }

  /* Check inventory weapons */
  int display_num = 1;
  for (int i = 0; i < INVENTORY_MAX_SLOTS; i++) {
    InventorySlot *slot = inventory_get_slot((Inventory *)&p->inventory, i);
    if (!slot || slot->type != SLOT_WEAPON)
      continue;

    Weapon *w = &slot->data.weapon;
    if (!weapon_can_upgrade(w))
      continue;

    printf("  %2d. ", display_num++);
    weapon_print(w);
    printf(" [Slot %d]\n", i);

    const UpgradeRecipe *recipe = get_recipe_for_weapon(w);
    if (recipe) {
      int can_upgrade = p->gold >= recipe->gold_cost &&
                        check_materials(&p->inventory, recipe);
      printf("      -> Tier %d (%s)\n", recipe->to_tier,
             can_upgrade ? "READY" : "missing materials");
    }
    found = 1;
  }

  if (!found) {
    printf("  No weapons can be upgraded.\n");
    printf("  (Collect materials from monsters or buy upgradeable weapons)\n");
  }

  printf("\n");
}

/* ============================================
 * PUBLIC FUNCTIONS
 * ============================================ */

int upgrade_can_upgrade_weapon(const Player *player, int weapon_slot) {
  if (!player)
    return 0;

  const Weapon *w;
  if (weapon_slot < 0) {
    /* Equipped weapon */
    if (!player->has_weapon)
      return 0;
    w = &player->equipped_weapon;
  } else {
    InventorySlot *slot =
        inventory_get_slot((Inventory *)&player->inventory, weapon_slot);
    if (!slot || slot->type != SLOT_WEAPON)
      return 0;
    w = &slot->data.weapon;
  }

  if (!weapon_can_upgrade(w))
    return 0;

  const UpgradeRecipe *recipe = get_recipe_for_weapon(w);
  if (!recipe)
    return 0;

  if (player->gold < recipe->gold_cost)
    return 0;
  if (!check_materials(&player->inventory, recipe))
    return 0;

  return 1;
}

ResultCode upgrade_weapon(Player *player, int weapon_slot) {
  if (!player)
    return RESULT_FAIL;

  Weapon *w;
  if (weapon_slot < 0) {
    if (!player->has_weapon)
      return RESULT_INVALID_SLOT;
    w = &player->equipped_weapon;
  } else {
    InventorySlot *slot = inventory_get_slot(&player->inventory, weapon_slot);
    if (!slot || slot->type != SLOT_WEAPON)
      return RESULT_INVALID_SLOT;
    w = &slot->data.weapon;
  }

  if (!weapon_can_upgrade(w)) {
    return RESULT_MAX_TIER_REACHED;
  }

  const UpgradeRecipe *recipe = get_recipe_for_weapon(w);
  if (!recipe) {
    return RESULT_CANNOT_UPGRADE;
  }

  /* Check gold */
  if (player->gold < recipe->gold_cost) {
    return RESULT_NOT_ENOUGH_GOLD;
  }

  /* Check materials */
  if (!check_materials(&player->inventory, recipe)) {
    return RESULT_NOT_ENOUGH_MATERIALS;
  }

  /* Perform upgrade */
  player_spend_gold(player, recipe->gold_cost);
  consume_materials(&player->inventory, recipe);
  weapon_apply_upgrade(w, recipe->damage_bonus);

  return RESULT_SUCCESS;
}

MenuResult upgrade_enter(Player *player) {
  if (!player)
    return MENU_BACK;

  while (1) {
    clear_screen();
    print_header("WEAPON UPGRADE", 50);
    printf("\n  Your Gold: %d\n\n", player->gold);

    print_upgradeable_weapons(player);

    /* Show materials */
    inventory_print_materials(&player->inventory);

    printf("\n--- Upgrade Menu ---\n");
    printf("1. Upgrade Equipped Weapon\n");
    printf("2. Upgrade Inventory Weapon\n");
    printf("3. View Details\n");
    printf("4. Back\n");
    printf("Choice: ");

    int choice = get_int_input(1, 4);

    switch (choice) {
    case 1: {
      /* Upgrade equipped */
      if (!player->has_weapon) {
        printf("No weapon equipped.\n");
        delay_ms(DELAY_MENU_TRANSITION);
        break;
      }

      printf("\n");
      upgrade_print_requirements(&player->equipped_weapon, player);
      printf("\nUpgrade?");

      if (get_yes_no()) {
        ResultCode result = upgrade_weapon(player, -1);
        switch (result) {
        case RESULT_SUCCESS:
          printf("\nUpgrade successful!\n");
          weapon_print(&player->equipped_weapon);
          printf("\n");
          break;
        case RESULT_NOT_ENOUGH_GOLD:
          printf("\nNot enough gold!\n");
          break;
        case RESULT_NOT_ENOUGH_MATERIALS:
          printf("\nNot enough materials!\n");
          break;
        case RESULT_MAX_TIER_REACHED:
          printf("\nWeapon is at max tier!\n");
          break;
        default:
          printf("\nUpgrade failed.\n");
        }
      }
      delay_ms(DELAY_MENU_TRANSITION);
      break;
    }

    case 2: {
      /* Upgrade from inventory */
      printf("\nEnter weapon slot number (0 to cancel): ");
      int slot_num = get_int_input(0, INVENTORY_MAX_SLOTS);

      if (slot_num == 0)
        break;

      /* Find actual slot */
      int actual_slot = -1;
      int display_num = 0;
      for (int i = 0; i < INVENTORY_MAX_SLOTS; i++) {
        InventorySlot *slot = inventory_get_slot(&player->inventory, i);
        if (slot && slot->type == SLOT_WEAPON &&
            weapon_can_upgrade(&slot->data.weapon)) {
          display_num++;
          if (display_num == slot_num) {
            actual_slot = i;
            break;
          }
        }
      }

      if (actual_slot < 0) {
        printf("Invalid slot.\n");
        delay_ms(DELAY_MENU_TRANSITION);
        break;
      }

      InventorySlot *slot = inventory_get_slot(&player->inventory, actual_slot);
      printf("\n");
      upgrade_print_requirements(&slot->data.weapon, player);
      printf("\nUpgrade?");

      if (get_yes_no()) {
        ResultCode result = upgrade_weapon(player, actual_slot);
        if (result == RESULT_SUCCESS) {
          printf("\nUpgrade successful!\n");
          weapon_print(&slot->data.weapon);
          printf("\n");
        } else {
          printf("\nUpgrade failed.\n");
        }
      }
      delay_ms(DELAY_MENU_TRANSITION);
      break;
    }

    case 3: {
      /* View details */
      printf("\nEnter 'E' for equipped or slot number: ");
      char input[10];
      if (scanf("%9s", input) != 1) {
        clear_input_buffer();
        break;
      }

      if (input[0] == 'E' || input[0] == 'e') {
        if (player->has_weapon) {
          printf("\n");
          weapon_print_detailed(&player->equipped_weapon);
          printf("\n");
          upgrade_print_requirements(&player->equipped_weapon, player);
        }
      }

      wait_for_enter();
      break;
    }

    case 4:
      return MENU_BACK;

    default:
      break;
    }
  }
}

File: internal/tui/screens/upgrade.go
// upgrade.go implements the weapon upgrade screen.
// Shows available upgrade recipes with material requirements, gold costs,
// and damage bonuses. Validates requirements before applying upgrades.
package screens

import (
	"fmt"
	"strings"

	"github.com/charmbracelet/bubbles/key"
	tea "github.com/charmbracelet/bubbletea"
	"github.com/tenyom/textrpg-tui/internal/domain"
	"github.com/tenyom/textrpg-tui/internal/infrastructure/registry"
	"github.com/tenyom/textrpg-tui/internal/tui/styles"
)

// Pre-allocated key bindings for the upgrade screen.
var (
	upgKeyUp    = key.NewBinding(key.WithKeys("up", "k"))
	upgKeyDown  = key.NewBinding(key.WithKeys("down", "j"))
	upgKeyEnter = key.NewBinding(key.WithKeys("enter"))
	upgKeyEsc   = key.NewBinding(key.WithKeys("esc", "backspace"))
)

// UpgradeMode represents the current upgrade submenu.
type UpgradeMode int

const (
	UpgradeModeMain UpgradeMode = iota
	UpgradeModeViewEquipped
	UpgradeModeViewInventory
	UpgradeModeDetails
)

// UpgradeScreen implements the weapon upgrade screen.
type UpgradeScreen struct {
	player   *domain.Player
	weapons  *registry.WeaponRegistry
	items    *registry.ItemRegistry
	upgrades *registry.UpgradeRegistry

	mode     UpgradeMode
	cursor   int
	width    int
	height   int
	message  string
	msgStyle string
}

// NewUpgradeScreen creates a new upgrade screen.
func NewUpgradeScreen(
	player *domain.Player,
	weapons *registry.WeaponRegistry,
	items *registry.ItemRegistry,
	upgrades *registry.UpgradeRegistry,
) *UpgradeScreen {
	return &UpgradeScreen{
		player:   player,
		weapons:  weapons,
		items:    items,
		upgrades: upgrades,
		mode:     UpgradeModeMain,
		cursor:   0,
	}
}

// Title returns the screen title.
func (u *UpgradeScreen) Title() string {
	return "WEAPON UPGRADE"
}

// SetSize updates dimensions.
func (u *UpgradeScreen) SetSize(width, height int) {
	u.width = width
	u.height = height
}

// Update handles upgrade screen input.
func (u *UpgradeScreen) Update(msg tea.Msg) tea.Cmd {
	switch msg := msg.(type) {
	case tea.KeyMsg:
		switch {
		case key.Matches(msg, upgKeyUp):
			if u.cursor > 0 {
				u.cursor--
			}
		case key.Matches(msg, upgKeyDown):
			u.cursor++
		case key.Matches(msg, upgKeyEnter):
			return u.handleSelect()
		case key.Matches(msg, upgKeyEsc):
			if u.mode != UpgradeModeMain {
				u.mode = UpgradeModeMain
				u.cursor = 0
				u.message = ""
			} else {
				return EndScreenCmd()
			}
		case msg.String() >= "1" && msg.String() <= "4":
			idx := int(msg.String()[0] - '1')
			u.cursor = idx
			return u.handleSelect()
		}
	}
	return nil
}

// handleSelect processes selection based on mode.
func (u *UpgradeScreen) handleSelect() tea.Cmd {
	switch u.mode {
	case UpgradeModeMain:
		switch u.cursor {
		case 0:
			u.doUpgradeEquipped()
		case 1:
			u.mode = UpgradeModeViewInventory
			u.cursor = 0
		case 2:
			u.mode = UpgradeModeDetails
			u.cursor = 0
		case 3: // Back
			return EndScreenCmd()
		}

	case UpgradeModeViewInventory:
		u.doUpgradeInventory()
	}
	return nil
}

// doUpgradeEquipped attempts to upgrade equipped weapon.
func (u *UpgradeScreen) doUpgradeEquipped() {
	if u.player.EquippedWeapon == nil {
		u.message = "No weapon equipped."
		u.msgStyle = "error"
		return
	}

	w := u.player.EquippedWeapon
	recipe := u.upgrades.GetRecipe(w.ID, w.Tier)

	if recipe == nil {
		if w.Tier >= w.MaxTier {
			u.message = "Weapon is at maximum tier!"
		} else {
			u.message = "No upgrade path available."
		}
		u.msgStyle = "error"
		return
	}

	// Check gold
	if u.player.Gold < recipe.GoldCost {
		u.message = fmt.Sprintf("Not enough gold! Need %d G", recipe.GoldCost)
		u.msgStyle = "error"
		return
	}

	// Check materials
	if !u.checkMaterials(recipe) {
		u.message = "Not enough materials!"
		u.msgStyle = "error"
		return
	}

	// Perform upgrade
	u.player.Gold -= recipe.GoldCost
	u.consumeMaterials(recipe)
	w.Tier = recipe.ToTier
	w.BaseDamage += recipe.DamageBonus

	u.message = fmt.Sprintf("Upgrade successful! %s +%d", w.Name, w.Tier-1)
	u.msgStyle = "success"
}

// doUpgradeInventory upgrades weapon from inventory.
func (u *UpgradeScreen) doUpgradeInventory() {
	upgradeables := u.getUpgradeableWeapons()
	if u.cursor < 0 || u.cursor >= len(upgradeables) {
		return
	}

	w := upgradeables[u.cursor].weapon
	recipe := u.upgrades.GetRecipe(w.ID, w.Tier)

	if recipe == nil {
		u.message = "No upgrade available."
		u.msgStyle = "error"
		return
	}

	if u.player.Gold < recipe.GoldCost {
		u.message = "Not enough gold!"
		u.msgStyle = "error"
		return
	}

	if !u.checkMaterials(recipe) {
		u.message = "Not enough materials!"
		u.msgStyle = "error"
		return
	}

	// Perform upgrade
	u.player.Gold -= recipe.GoldCost
	u.consumeMaterials(recipe)
	w.Tier = recipe.ToTier
	w.BaseDamage += recipe.DamageBonus

	u.message = fmt.Sprintf("Upgrade successful! %s +%d", w.Name, w.Tier-1)
	u.msgStyle = "success"
}

// checkMaterials checks if player has required materials.
func (u *UpgradeScreen) checkMaterials(recipe *registry.UpgradeRecipe) bool {
	for _, mat := range recipe.Materials {
		have := u.player.Inventory.CountItem(mat.ItemID)
		if have < mat.Quantity {
			return false
		}
	}
	return true
}

// consumeMaterials removes materials from inventory.
func (u *UpgradeScreen) consumeMaterials(recipe *registry.UpgradeRecipe) {
	for _, mat := range recipe.Materials {
		u.player.Inventory.RemoveItemByID(mat.ItemID, mat.Quantity)
	}
}

type upgradeableEntry struct {
	weapon   *domain.Weapon
	slotIdx  int
	equipped bool
}

// getUpgradeableWeapons returns weapons that can be upgraded.
func (u *UpgradeScreen) getUpgradeableWeapons() []upgradeableEntry {
	var result []upgradeableEntry

	// Check equipped
	if u.player.EquippedWeapon != nil && u.player.EquippedWeapon.CanUpgrade() {
		result = append(result, upgradeableEntry{
			weapon:   u.player.EquippedWeapon,
			slotIdx:  -1,
			equipped: true,
		})
	}

	// Check inventory
	for i := 0; i < domain.MaxInventorySlots; i++ {
		slot := u.player.Inventory.GetSlot(i)
		if slot != nil && slot.Type == domain.SlotWeapon {
			if slot.Weapon.CanUpgrade() {
				result = append(result, upgradeableEntry{
					weapon:   slot.Weapon,
					slotIdx:  i,
					equipped: false,
				})
			}
		}
	}

	return result
}

// View renders the upgrade screen.
// FORMAT MATCHES C src/game/upgrade.c upgrade_enter()
func (u *UpgradeScreen) View() string {
	var b strings.Builder

	// Header matching C: print_header("WEAPON UPGRADE", 50)
	b.WriteString(styles.SectionHeaderStyle.Render("=== WEAPON UPGRADE ==="))
	b.WriteString("\n\n")

	// Gold display
	goldLabel := styles.StatLabelStyle.Render("  Your Gold: ")
	goldValue := styles.GoldStyle.Render(fmt.Sprintf("%d G", u.player.Gold))
	b.WriteString(goldLabel + goldValue)
	b.WriteString("\n\n")

	// Content based on mode
	switch u.mode {
	case UpgradeModeMain:
		b.WriteString(u.renderUpgradeableWeapons())
		b.WriteString("\n")
		b.WriteString(u.renderMaterials())
		b.WriteString("\n")
		b.WriteString(u.renderMainMenu())
	case UpgradeModeViewInventory:
		b.WriteString(u.renderInventoryWeapons())
	case UpgradeModeDetails:
		b.WriteString(u.renderDetails())
	}

	// Status message
	if u.message != "" {
		b.WriteString("\n")
		switch u.msgStyle {
		case "success":
			b.WriteString(styles.SuccessStyle.Render(u.message))
		case "error":
			b.WriteString(styles.ErrorStyle.Render(u.message))
		default:
			b.WriteString(u.message)
		}
	}

	return b.String()
}

// renderUpgradeableWeapons matches C print_upgradeable_weapons().
func (u *UpgradeScreen) renderUpgradeableWeapons() string {
	var b strings.Builder

	b.WriteString(styles.SectionHeaderStyle.Render("=== Upgradeable Weapons ==="))
	b.WriteString("\n\n")

	upgradeables := u.getUpgradeableWeapons()

	if len(upgradeables) == 0 {
		b.WriteString(styles.DisabledItemStyle.Render("  No weapons can be upgraded."))
		b.WriteString("\n")
		b.WriteString(styles.DisabledItemStyle.Render("  (Collect materials from monsters or buy upgradeable weapons)"))
		return b.String()
	}

	for i, entry := range upgradeables {
		w := entry.weapon
		recipe := u.upgrades.GetRecipe(w.ID, w.Tier)

		var line strings.Builder
		if entry.equipped {
			line.WriteString("  [E] ")
		} else {
			line.WriteString(fmt.Sprintf("  %2d. ", i+1))
		}

		line.WriteString(w.DisplayName())
		if entry.equipped {
			line.WriteString(" (EQUIPPED)")
		}
		b.WriteString(styles.StatValueStyle.Render(line.String()))
		b.WriteString("\n")

		// Show upgrade path
		if recipe != nil {
			canUpgrade := u.player.Gold >= recipe.GoldCost && u.checkMaterials(recipe)
			status := "missing materials"
			if canUpgrade {
				status = "READY"
			}
			b.WriteString(fmt.Sprintf("      -> Tier %d (%s)\n", recipe.ToTier, status))
		}
	}

	return b.String()
}

// renderMaterials shows player's materials.
func (u *UpgradeScreen) renderMaterials() string {
	var b strings.Builder

	b.WriteString(styles.SectionHeaderStyle.Render("=== Materials ==="))
	b.WriteString("\n")

	materials := u.player.Inventory.GetMaterials()
	if len(materials) == 0 {
		b.WriteString(styles.DisabledItemStyle.Render("  No materials"))
		return b.String()
	}

	for _, mat := range materials {
		b.WriteString(fmt.Sprintf("  %s x%d\n", mat.Name, mat.Quantity))
	}

	return b.String()
}

// renderMainMenu matches C upgrade menu.
func (u *UpgradeScreen) renderMainMenu() string {
	var b strings.Builder

	b.WriteString(styles.SeparatorStyle.Render("--- Upgrade Menu ---"))
	b.WriteString("\n")

	items := []string{
		"1. Upgrade Equipped Weapon",
		"2. Upgrade Inventory Weapon",
		"3. View Details",
		"4. Back",
	}

	for i, item := range items {
		if i == u.cursor && u.mode == UpgradeModeMain {
			b.WriteString(styles.SelectedItemStyle.Render(item))
		} else {
			b.WriteString(styles.NormalItemStyle.Render(item))
		}
		b.WriteString("\n")
	}

	b.WriteString("Choice: ")
	return b.String()
}

// renderInventoryWeapons shows inventory weapons for upgrade selection.
func (u *UpgradeScreen) renderInventoryWeapons() string {
	var b strings.Builder

	b.WriteString(styles.SectionHeaderStyle.Render("=== Select Weapon to Upgrade ==="))
	b.WriteString("\n\n")

	upgradeables := u.getUpgradeableWeapons()
	if u.cursor >= len(upgradeables) && len(upgradeables) > 0 {
		u.cursor = len(upgradeables) - 1
	}

	for i, entry := range upgradeables {
		w := entry.weapon
		recipe := u.upgrades.GetRecipe(w.ID, w.Tier)

		var line strings.Builder
		line.WriteString(fmt.Sprintf("  %2d. %s", i+1, w.DisplayName()))
		if entry.equipped {
			line.WriteString(" (EQUIPPED)")
		}

		if recipe != nil {
			line.WriteString(fmt.Sprintf(" [%d G]", recipe.GoldCost))
		}

		if i == u.cursor {
			b.WriteString(styles.SelectedItemStyle.Render(line.String()))
		} else {
			b.WriteString(line.String())
		}
		b.WriteString("\n")

		// Show requirements for selected
		if i == u.cursor && recipe != nil {
			b.WriteString(u.renderRecipeRequirements(recipe))
		}
	}

	b.WriteString("\n")
	b.WriteString(styles.HelpDescStyle.Render("[Enter] Upgrade  [Esc] Back"))

	return b.String()
}

// renderRecipeRequirements shows materials needed for upgrade.
func (u *UpgradeScreen) renderRecipeRequirements(recipe *registry.UpgradeRecipe) string {
	var b strings.Builder

	b.WriteString("      Requirements:\n")

	// Gold
	goldOK := u.player.Gold >= recipe.GoldCost
	goldStatus := "[OK]"
	if !goldOK {
		goldStatus = fmt.Sprintf("[NEED %d more]", recipe.GoldCost-u.player.Gold)
	}
	b.WriteString(fmt.Sprintf("        Gold: %d %s\n", recipe.GoldCost, goldStatus))

	// Materials
	for _, mat := range recipe.Materials {
		have := u.player.Inventory.CountItem(mat.ItemID)
		need := mat.Quantity

		matName := mat.ItemID
		if item := u.items.GetByID(mat.ItemID); item != nil {
			matName = item.Name
		}

		status := "[OK]"
		if have < need {
			status = fmt.Sprintf("[NEED %d more]", need-have)
		}
		b.WriteString(fmt.Sprintf("        %s: %d/%d %s\n", matName, have, need, status))
	}

	return b.String()
}

// renderDetails shows detailed weapon info.
func (u *UpgradeScreen) renderDetails() string {
	var b strings.Builder

	b.WriteString(styles.SectionHeaderStyle.Render("=== Weapon Details ==="))
	b.WriteString("\n\n")

	if u.player.EquippedWeapon != nil {
		w := u.player.EquippedWeapon
		b.WriteString(fmt.Sprintf("  %s (EQUIPPED)\n", w.DisplayName()))
		b.WriteString(fmt.Sprintf("  Type: %s\n", w.Type.String()))
		b.WriteString(fmt.Sprintf("  Element: %s\n", w.Element.String()))
		b.WriteString(fmt.Sprintf("  Damage: %d\n", w.Damage()))
		b.WriteString(fmt.Sprintf("  Tier: %d/%d\n", w.Tier, w.MaxTier))

		recipe := u.upgrades.GetRecipe(w.ID, w.Tier)
		if recipe != nil {
			b.WriteString(fmt.Sprintf("  Next Tier Damage: +%d\n", recipe.DamageBonus))
		}
	} else {
		b.WriteString(styles.DisabledItemStyle.Render("  No weapon equipped"))
	}

	b.WriteString("\n")
	b.WriteString(styles.HelpDescStyle.Render("[Esc] Back"))

	return b.String()
}
